src/GafferScene/Render.cpp

```diff
@@ -336,16 +337,21 @@ void Render::executeInternal( bool flushCaches ) const
 	if( flushCaches )
 	{
 		// Now we have generated the scene, flush Cortex and Gaffer caches to
-		// provide more memory to the renderer.
-		/// \todo If executing directly within the gui app flushing the caches
-		/// is definitely not wanted. Since this scenario is currently uncommon,
-		/// we prioritise the common case of performing a single render from within
-		/// `gaffer execute`, but it would be good to do better.
-		ObjectPool::defaultObjectPool()->clear();
-		ValuePlug::clearCache();
-		/// \todo This is not as effective as it could be, because it doesn't actually
-		/// clear the per-thread caches until the next operation on each thread.
-		ValuePlug::clearHashCache();
+		// provide more memory to the renderer. We limit this to the `execute`
+		// and `dispatch` applications for two reasons :
+		//
+		// - In a GUI application, we don't want to clear the caches because
+		//   we'll probably benefit from using them again later.
+		// - In `execute` and `dispatch` we know we're not executing concurrently
+		//   with anything else, and can therefore pass `now = true` to
+		//   `clearHashCache()` safely.
+		auto *application = ancestor<ApplicationRoot>();
+		if( application && ( application->getName() == "execute" || application->getName() == "dispatch" ) )
```

Comment:
We are just about to introduce a new application that executes tasks (`caribouNukeExecute`), which would likely benefit from the same behaviour.

Could we make the list of app names configurable somehow?

Reply:
This comment has come a bit late given that we planned to release today, so I'm going to go ahead and merge anyway. But we can absolutely discuss a configuration mechanism you can use for `caribouNukeExecute`, and get that into a subsequent release.

Out of interest, how does `caribouNukeExecute` differ from `execute`? Does it indicate some lack of flexibility in the default functionality?

Reply:
`caribouNukeExecute` is almost a copy of the `execute` app with the main difference being that it starts a host ( nuke but it could be something else if we wanted to ) as a new sub process and then load the application ( itself ) once the host is running and finally run the Gaffer `TaskNode`s connected to a special `TaskPlug` inside the `CaribouNuke` Gaffer graph.

Let me know if you want to discuss that topic in more details

Reply:
Ah, cool, then that's a bit out of scope for what we intend the `execute` app to be able to handle, so I think a custom app is the way to go. Actually, we'd love to provide host-specific dispatchers so GafferDispatch could be used for task orchestration in any host, but we don't have the stomach for managing the hell of making official builds that are ABI compatible with all the versions of all the DCCs.

I'll put a bit of thought into a configuration mechanism and then get back to you - when would you need it by?